src/map/skill.c

```diff
@@ -17051,6 +17055,13 @@ static struct skill_unit *skill_initunit(struct skill_unit_group *group, int idx
 	su->val1=val1;
 	su->val2 = val2;
 	su->prev = 0;
+	su->visible = true;
+
+	if (skill->get_inf2(group->skill_id) & INF2_HIDDEN_TRAP
+		&& ((battle_config.trap_visibility == 1 && map_flag_vs(group->map)) // invisible in PvP/GvG
+			|| battle_config.trap_visibility == 2 // always invisible
+		))
+	 	su->visible = false;
```

Comment:
for complex if like this, better add ``{}`` before/after code inside if block.
``su->visible = false;`` -->
```
{
    su->visible = false;
}
```